src/microbe_stage/MicrobeSystem.cs

```diff
@@ -54,4 +57,63 @@ public void Process(float delta)
             microbe.ProcessSync(delta);
         }
     }
+
+    /// <summary>
+    ///   Tries to find specified Species as close to the point as possible.
+    /// </summary>
+    /// <param name="position">Position to search around</param>
+    /// <param name="species">What species to search for</param>
+    /// <param name="searchRadius">How wide to search around the point</param>
+    /// <returns>The nearest found point for the species or null</returns>
+    public (Microbe Microbe, Vector3 Position)? FindSpeciesNearPoint(Vector3 position, Species species,
+        float searchRadius = 200)
+    {
+        if (searchRadius < 1)
+            throw new ArgumentException("searchRadius must be >= 1");
+
+        (Microbe Microbe, Vector3 Position)? closestMicrobe = null;
+        float nearestDistanceSquared = float.MaxValue;
+        var searchRadiusSquared = searchRadius * searchRadius;
+        microbes ??= worldRoot.GetTree().GetNodesInGroup(Constants.RUNNABLE_MICROBE_GROUP).Cast<Microbe>()
+            .ToArray();
+
+        foreach (var microbe in microbes)
+        {
+            if (microbe.Species != species)
+                continue;
+
+            Vector3 microbeGlobalPosition;
+
+            // Use colony parent position to avoid calling GlobalTranslation
+            if (microbe.ColonyParent != null)
+            {
+                microbeGlobalPosition = microbe.ColonyParent.Translation;
+            }
+            else
+            {
+                microbeGlobalPosition = microbe.Translation;
+            }
+
+            // Skip candidates for performance
+            if (Math.Abs(microbeGlobalPosition.x - position.x) > searchRadius ||
+                Math.Abs(microbeGlobalPosition.y - position.y) > searchRadius)
+            {
+                continue;
+            }
+
+            var distanceSquared = (microbeGlobalPosition - position).LengthSquared();
+
+            // Check search range and prevent cell detecting itself
+            if (distanceSquared > searchRadiusSquared || distanceSquared <= 1)
+                continue;
+
+            if (distanceSquared < nearestDistanceSquared)
```

Comment:
These could be maybe swapped around to simplify the logic here. It probably won't perform worse as the microbes are very likely not in a descending order of distance to the target position. And it would entirely remove the need to check if the search radius is followed here. That can be added just at the end to return null instead of the closest microbe if it is farther than the search radius.